### PR TITLE
Enable new trees

### PR DESCRIPTION
**Pull request by musamaanjum**
Enable the following trees:
- agross
- khilman
- kselftest
- mainline

### CODE REVIEW

File: config/pipeline.yaml [hunk 1]
```diff
@@ -449,6 +527,29 @@ jobs:
...
 
+  kbuild-gcc-12-arc-mainline: &kbuild-gcc-12-arc-mainline-job
+    <<: *kbuild-gcc-12-arc-job
+    params: &kbuild-gcc-12-arc-mainline-params
+      <<: *kbuild-gcc-12-arc-params
+      fragments:
+        - 'debug'
+        - 'kselftest'
+        - 'tinyconfig'
+    rules:
```
Comment (musamaanjum) [inline config/pipeline.yaml:538]: These rules have been translated from
```
            - blocklist:
                defconfig:
                  - 'axs101_defconfig'
                  - 'nps_defconfig'
                  - 'nsim_700_defconfig'
                  - 'nsimosci_defconfig'
                  - 'tb10x_defconfig'
            # haps_hs_smp_defconfig not available in some old kernels
                kernel: ['v3.', 'v4.4', 'v4.9']
```
Please confirm if I've translated correctly:
- I think we don't need to add the rule to exclude the mentioned configs as we aren't enabling them.
- I've enabled builds for kernel versions starting from 4.10.

File: config/pipeline.yaml [hunk 2]
```diff
@@ -773,6 +1039,29 @@ jobs:
...
 
+  kbuild-gcc-12-riscv-mainline: &kbuild-gcc-12-riscv-mainline-job
+    <<: *kbuild-gcc-12-riscv-job
+    params: &kbuild-gcc-12-riscv-mainline-params
+      <<: *kbuild-gcc-12-riscv-params
+      fragments:
+        - 'debug'
+        - 'kselftest'
+        - 'tinyconfig'
+    rules:
```
Comment (musamaanjum) [inline config/pipeline.yaml:1050]: From old code;
```
            - blocklist: &riscv_kernel_filter
                kernel: ['v3.', 'v4.4', 'v4.9', 'v4.14']
```
I've enabled riscv builds from v4.15. Does it seem correct?

Comment (nuclearcat) [reply config/pipeline.yaml:1050]: Correct

File: config/pipeline.yaml (hunk 1 above)
Comment (nuclearcat) [reply config/pipeline.yaml:538]: Yes, correct

**Comment by musamaanjum** [comment]
Results:

agross:
https://staging.kernelci.org:9000/viewer?node_id=66d737a9054960692a2409b5
https://staging.kernelci.org:9000/viewer?node_id=66d737ab054960692a2409b9
khilman:
https://staging.kernelci.org:9000/viewer?node_id=66d73995054960692a240a55
https://staging.kernelci.org:9000/viewer?node_id=66d73995054960692a240a56
kselftest:
https://staging.kernelci.org:9000/viewer?node_id=66d739ee054960692a240bb3
https://staging.kernelci.org:9000/viewer?node_id=66d739ef054960692a240bb4
https://staging.kernelci.org:9000/viewer?node_id=66d739f2054960692a240bb8
mainline:
https://staging.kernelci.org:9000/viewer?node_id=66ca78c545d2bdb448d30291
https://staging.kernelci.org:9000/viewer?node_id=66ca78c645d2bdb448d30292
https://staging.kernelci.org:9000/viewer?node_id=66ca78c745d2bdb448d30293

**Comment by nuclearcat** [comment]
Thanks a lot Usama!

**Pull request merged**